### ISSUE TITLE
skbio.io.read: improve error message when format has no generator reader

### ISSUE THREAD

**Issue by jairideout** [biocore/scikit-bio#1326]
When reading a file using the procedural `skbio.io.read` interface, the error message could be improved when `into` isn't provided and there isn't a generator reader. For example:

```python
import skbio.io
msa = skbio.io.read('my-stockholm-file', format='stockholm')
...
UnrecognizedFormatError: Cannot read 'stockholm' from <_io.BufferedReader name='my-stockholm-file'>, no generator reader found.
```

It'd be helpful to provide a hint about passing `into`, including valid classes for the provided `format`. In this example (Stockholm), `TabularMSA` is the only class the file can be read into.

**Comment by ebolyen** [comment]
:+1:

**Comment by jairideout** [comment]
@Kleptobismol ping on this issue, would be nice to have for 0.5.0 (release scheduled for Friday 6/10). The `StopIteration` error you were receiving should now be fixed upstream (#1369).
